src/ResourceManager/CognitiveServices/CognitiveServices.Test/ScenarioTests/Common.ps1

```diff
@@ -0,0 +1,141 @@
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+Cleans the created resource groups
+#>
+function Clean-ResourceGroup($rgname)
+{
+    if ([Microsoft.Azure.Test.HttpRecorder.HttpMockServer]::Mode -ne [Microsoft.Azure.Test.HttpRecorder.HttpRecorderMode]::Playback) {
+        Remove-AzureRmResourceGroup -Name $rgname -Force
+    }
+}
+
+######################
+#
+# Retry the given code block until it succeeds or times out.
+#
+#    param [ScriptBlock] $script : The code to test
+#    param [int] $times          : The times of running the code
+#    param [string] $message     : The text of the exception that should be thrown
+#######################
+function Retry-IfException
+{
+    param([ScriptBlock] $script, [int] $times = 30, [string] $message = "*")
+
+    if ($times -le 0)
+    {
+        throw 'Retry time(s) should not be equal to or less than 0.';
+    }
+
+    $oldErrorActionPreferenceValue = $ErrorActionPreference;
+    $ErrorActionPreference = "SilentlyContinue";
+
+    $iter = 0;
+    $succeeded = $false;
+    while (($iter -lt $times) -and (-not $succeeded))
+    {
+        $iter += 1;
+
+        try
+        {
+            &$script;
+        }
+        catch
+        {
+
+        }
+
+        if ($Error.Count -gt 0)
+        {
+            $actualMessage = $Error[0].Exception.Message;
+
+            Write-Output ("Caught exception: '$actualMessage'");
+
+            if (-not ($actualMessage -like $message))
+            {
+                $ErrorActionPreference = $oldErrorActionPreferenceValue;
+                throw "Expected exception not received: '$message' the actual message is '$actualMessage'";
+            }
+
+            $Error.Clear();
+            Wait-Seconds 10;
```

Comment:
This should be conditioned on !TestSupport.RunningMocked, to avoid unnecessary waits when running in playback mode